ui/src/app/workflows/components/workflow-logs-viewer/workflow-logs-viewer.tsx

```diff
@@ -18,7 +18,8 @@ interface WorkflowLogsViewerProps {
 interface WorkflowLogsViewerState {
     error?: Error;
     loaded: boolean;
-    lines: string[];
+    lineCount: number;
+    logsObservable?: Observable<string>;
```

Comment:
should this be part of the state? or just a field?

Reply:
So this is where my knowledge of React falls flat a bit (historically most of my development is in Vue/Angular).  I put this object into the `state` only because it's referenced from within JSX.  I'm not entirely sure whether that's the right thing to do or not.  I can try using this as a simple field and see if that works as well.  I *think* it will since it doesn't directly participate in data flow (it's returned via a function).

Reply:
but it works?

Reply:
Seems to, just tried it.  I'll make that change in a couple minutes.